ugvc/methylation/process_mergeContext.py

```
@@ -0,0 +1,306 @@
+
+                        # add other metrics to absolute values use x from above
+                        desc = x.describe()
+                        desc = desc[rows]
+                        desc = desc.rename(index={'50%': 'median'})
+                        desc = pd.DataFrame({'metric': desc.index, 'value': desc})
+                        df_abs_cov = (pd.concat([df_abs_cov, desc], axis=0, ignore_index=True))
+                        # assign names for metrics
+                        df_abs_cov['metric'] = col + "_" + df_abs_cov['metric'].astype(str)
+                        col = 'CumulativeCoverage'
+                        df_rel_cov['metric'] = col + "_" + df_rel_cov['metric'].astype(str)
+
+                df_distrib = df_distrib.append([df_pcnt, df_abs_cov, df_rel_cov], ignore_index=True)
+                df_distrib.fillna(0, inplace=True)
+                df_distrib['detail'] = key_word
+                return (df_distrib)
+
+
+        # function for extracting chromosomes and genomes from input file
+        # ===================================================================
+        def find_list(in_list):
+
+            string = ''.join(in_list)
+            match = re.findall(r"chr|Lambda|pUC19", string, re.IGNORECASE)
+            match = list(Counter(match))
+            return match
+
+
+        # Get chromosomes and genomes from MethylDackel mergeContext file
+        # ===================================================================
+        list_chroms_genomes = list(set(df_in_report.chr))
+        list_genomes = find_list(list_chroms_genomes)
+        list_genomes = [s.replace('chr', 'hg') for s in list_genomes]
+
+
+        # create dataframe of % methylation and coverage for each genome in MethylDackel output file
+        # ==========================================================================================
+        df_csv_output = pd.DataFrame()
+        # list_genomes: runs on [hg] or ['hg', 'Lambda', 'pUC19']

```

Are ['hg', 'Lambda', 'pUC19'] the only allowed values? If so I would raise an error if any genome is not in this list.

---

It can be [hg] or ['hg', 'Lambda', 'pUC19']
It runs on either options.